### PR TITLE
Sensitive media

### LINKED ISSUE #158: As a user, would like to have sensitive media hidden

### PR DESCRIPTION
**Pull request by SoupeDeKiwis**
Hide sensitive content based on 'sensitive' attribute from Status object.

### CODE REVIEW

File: app/src/androidTest/java/com/h/pixeldroid/MockedServerTest.kt [hunk 1]
```diff
@@ -501,5 +516,71 @@ class MockedServerTest {
...
+        onView(withId(R.id.list)).perform(slowSwipeUp(false))
+        onView(withId(R.id.list)).perform(slowSwipeUp(false))
+        Thread.sleep(1000)
+
+        onView(withId(R.id.list))
+            .perform(actionOnItemAtPosition<PostViewHolder>
+                (1, clickChildViewWithId(R.id.postPicture)))
+        Thread.sleep(1000)
+
+        assert(true)
```
Comment (dobios) [inline app/src/androidTest/java/com/h/pixeldroid/MockedServerTest.kt]: Try testing something in your tests please, like if the post has actually been shown or not.

File: app/src/androidTest/java/com/h/pixeldroid/MockedServerTest.kt [hunk 2]
```diff
@@ -501,5 +516,71 @@ class MockedServerTest {
...
+        onView(withId(R.id.list)).perform(slowSwipeUp(false))
+        onView(withId(R.id.list)).perform(slowSwipeUp(false))
+        Thread.sleep(1000)
+
+        onView(withId(R.id.list))
+            .perform(actionOnItemAtPosition<PostViewHolder>
+                (1, clickChildViewWithId(R.id.sensitiveWarning)))
+        Thread.sleep(1000)
+
+        assert(true)
```
Comment (dobios) [inline app/src/androidTest/java/com/h/pixeldroid/MockedServerTest.kt]: The test should check that the click works. No assert(true) please

**Review by dobios — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/com/h/pixeldroid/MockedServerTest.kt (hunk 1 above)
Comment (SoupeDeKiwis) [reply app/src/androidTest/java/com/h/pixeldroid/MockedServerTest.kt]: Commit : e93fdff

File: app/src/androidTest/java/com/h/pixeldroid/MockedServerTest.kt (hunk 2 above)
Comment (SoupeDeKiwis) [reply app/src/androidTest/java/com/h/pixeldroid/MockedServerTest.kt]: Commit : e93fdff

File: app/src/main/java/com/h/pixeldroid/objects/Status.kt [hunk 3]
```diff
@@ -407,4 +425,31 @@ data class Status(
...
+        fun uncensorPicture(view: View) {
+            if (!media_attachments.isNullOrEmpty()) {
+                view.sensitiveWarning.visibility = GONE
+                view.postPicture.colorFilter = ColorMatrixColorFilter(uncensorColorMatrix())
+
+                if (media_attachments.size > 1)
+                    setupTabsLayout(view, request, homeFragment)
+            }
+                imagePopUpMenu(view, homeFragment.requireActivity())
+            }
```
Comment (ulysse42) [inline app/src/main/java/com/h/pixeldroid/objects/Status.kt]: argh l'indentation

**Review by dobios — CHANGES_REQUESTED** [review]
Sensitive posts aren't hidden when going to the user's profile, this potentially defeats the purpose of even having a sensitive media flag no?

File: app/src/main/java/com/h/pixeldroid/objects/Status.kt (hunk 3 above)
Comment (SoupeDeKiwis) [reply app/src/main/java/com/h/pixeldroid/objects/Status.kt]: 3f4edc1

**Review by ulysse42 — APPROVED** [review]

**Review by dobios — APPROVED** [review]

**Review by Wv5twkFEKh54vo4tta9yu7dHa3 — CHANGES_REQUESTED** [review]
Move hardcoded strings to xml

**Review by Wv5twkFEKh54vo4tta9yu7dHa3 — APPROVED** [review]

**Pull request merged**